src/backend/XmlHttpRequest.ts

```diff
@@ -29,25 +29,32 @@ export default class XmlHttpRequest extends file_system.BaseFileSystem implement
   public prefixUrl: string;
   /**
    * Constructs the file system.
-   * @param listingUrl The path to the JSON file index generated by
+   * @param listingUrlOrObj index object or the path to the JSON file index generated by
    *   tools/XHRIndexer.coffee. This can be relative to the current webpage URL
    *   or absolutely specified.
    * @param prefixUrl The url prefix to use for all web-server requests.
    */
-  constructor(listingUrl: string, prefixUrl: string = '') {
+  constructor(listingUrlOrObj: string | Object, prefixUrl: string = '') {
     super();
-    if (listingUrl == null) {
-      listingUrl = 'index.json';
+    if (listingUrlOrObj == null) {
```

Comment:
I used to do `==` as a shortcut for `=== null || === undefined`, but now use `!` as it's shorter & lints better.

No need to change if you don't feel like it.